pkg/lib/interaction/intents/change_password.go

```diff
@@ -45,9 +45,23 @@ func (i *IntentChangePassword) DeriveEdgesForNode(graph *interaction.Graph, node
 			},
 		}, nil
 	case *nodes.NodeChangePasswordEnd:
-		// Password was not changed, ends the interaction
-		return nil, nil
+		return []interaction.Edge{
+			&nodes.EdgeDoEnsureSession{
+				Mode: nodes.EnsureSessionModeNoop,
+			},
+		}, nil
```

Comment:
Why do we need to add EdgeDoEnsureSession here?

Reply:
Currently it's possible to change password to same as previous value. Or should it be treated as fail for settings action?

Reply:
I think it should be a success, but the developer could enable password history check, which should cause the flow to return api error. That should be handled by password policy already.

Reply:
Does this work? `EdgeDoEnsureSession` should be irrelevant here.

```
case *nodes.NodeChangePasswordEnd:
  // Password was not changed
  return []interaction.Edge{&nodes.EdgeSettingsActionEnd{}}, nil
case *nodes.NodeDoUpdateAuthenticator:
  return []interaction.Edge{&nodes.EdgeSettingsActionEnd{}}, nil
case *nodes.NodeSettingsActionEnd:
  // Intent is finished.
  return nil, nil
```

Reply:
Should still need, similar to `IntentReauthenticate`. Tried removing it results in invalid `authentication required`